### PR TITLE
Refactor properties

### PR DESCRIPTION
**Pull request by ChloeDawn**
- Mapped all parameters
- Renamed `create` to `of`
- Renamed `IntegerProperty` to `IntProperty`
- Cleaned up constant names in `Properties`
- Renamed `getValueAsString` to `getName`
- Renamed `getValueClass` to `getValueType`

### CODE REVIEW

File: mappings/net/minecraft/state/property/IntProperty.mapping [hunk 1]
```diff
@@ -0,0 +1,12 @@
+CLASS bwk net/minecraft/state/property/IntProperty
```
Comment (asiekierka) [inline mappings/net/minecraft/state/property/IntProperty.mapping:1]: It's a property of Integer objects, not so much primitive ints.

File: mappings/net/minecraft/state/property/Property.mapping [hunk 2]
```diff
@@ -1,6 +1,8 @@
 CLASS bwn net/minecraft/state/property/Property
 	METHOD a getName ()Ljava/lang/String;
-	METHOD a getValueAsString (Ljava/lang/Comparable;)Ljava/lang/String;
-	METHOD b getValueClass ()Ljava/lang/Class;
+	METHOD a getName (Ljava/lang/Comparable;)Ljava/lang/String;
```
Comment (asiekierka) [inline mappings/net/minecraft/state/property/Property.mapping:3]: That method is clearly for getting the name of a value, IIRC, and should not be confused with getName(). As such, I'm saying no to this rename

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 3]
```diff
@@ -2,23 +2,23 @@ CLASS bwb net/minecraft/state/property/Properties
...
 	FIELD C WATERLOGGED Lbwc;
-	FIELD D AXIS_XZ Lbwi;
-	FIELD E AXIS_XYZ Lbwi;
-	FIELD F UP_BOOL Lbwc;
-	FIELD G DOWN_BOOL Lbwc;
-	FIELD H NORTH_BOOL Lbwc;
-	FIELD I EAST_BOOL Lbwc;
-	FIELD J SOUTH_BOOL Lbwc;
-	FIELD K WEST_BOOL Lbwc;
+	FIELD D HORIZONTAL_AXIS Lbwi;
```
Comment (liach) [inline mappings/net/minecraft/state/property/Properties.mapping:5]: Plural of `AXIS` is `AXES`, so `HORIZONTAL_AXES`. Same below.

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 4]
```diff
@@ -83,3 +83,7 @@ CLASS bwb net/minecraft/state/property/Properties
 	FIELD x SHORT Lbwc;
 	FIELD y SIGNAL_FIRE Lbwc;
 	FIELD z SNOWY Lbwc;
+	METHOD a (Lbwo;)Z
```
Comment (liach) [inline mappings/net/minecraft/state/property/Properties.mapping:86]: what does this boolean return value mean?

File: mappings/net/minecraft/state/property/Property.mapping (hunk 2 above)
Comment (ChloeDawn) [reply mappings/net/minecraft/state/property/Property.mapping:3]: `property.getName(value)` is perfectly clear as to its intent imo.

Comment (liach) [reply mappings/net/minecraft/state/property/Property.mapping:3]: `getSerializedName`? Referring to that `StringRepresentable` horror.

Comment (liach) [reply mappings/net/minecraft/state/property/Property.mapping:3]: Or `nameValue`?

File: mappings/net/minecraft/state/property/IntProperty.mapping (hunk 1 above)
Comment (ChloeDawn) [reply mappings/net/minecraft/state/property/IntProperty.mapping:1]: The constructor takes unboxed ints and every use of the property handles values as unboxed ints. This is a cleaner name, and the verbosity of Integer over Int is redundant imo.

File: mappings/net/minecraft/state/property/Property.mapping (hunk 2 above)
Comment (ChloeDawn) [reply mappings/net/minecraft/state/property/Property.mapping:3]: Imo this method should be the same name as the property name getter. Both return a name. So if we rename this method the other should be reconsidered too. I personally think getName is fine though, anything else appended is just redundant verbosity.

File: mappings/net/minecraft/state/property/Properties.mapping (hunk 3 above)
Comment (ChloeDawn) [reply mappings/net/minecraft/state/property/Properties.mapping:5]: It represents a key for a singular value: `state.get(AXIS)`. If you argue that this should be plural you should argue that FACING should be FACINGS too. But it is incorrect naming for such a property imo.

File: mappings/net/minecraft/state/property/Properties.mapping (hunk 4 above)
Comment (ChloeDawn) [reply mappings/net/minecraft/state/property/Properties.mapping:86]: It is a synthetic method for a predicate.

File: mappings/net/minecraft/state/property/Properties.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/state/property/Properties.mapping:5]: nice new name! old name is confusing as i thought it was a swizzle. please mark this as resolved.

**Review by liach — APPROVED** [review]

**Pull request merged**